File: prjs/prog/prblm/clrs/csharp/CLRS.Tests/StringMatchingTest.cs
namespace CLRS.Tests;
using CLRS.Part7.Chap32;

public class StringMatchingTest
{
    [Fact]
    public void TestRabinKarp()
    {
         Assert.True(StringMatching.RabinKarp("nitin","nitin") == 0," should be true");
    }

    [Fact]
    public void TestKnuthMorisPratt(){
         Assert.True(StringMatching.KnuthMorrisPratt("nitin","nitin") == 0,"shoudl be true");
    }
}

File: prjs/prog/prblm/clrs/cpp/clrs/part7/chap32/string_matching_test.cpp
#include "string_matching.h"
#include<iostream>
using namespace std;
using namespace part7::chap32;

int main(){
    auto idx = StringMatching::RabinKarp(
        "nitin",
        "nnitin"
    );
    
    cout<<" Runing test "<<idx;
    return 0;
}